homeassistant/components/nest/climate_sdm.py

```diff
@@ -0,0 +1,345 @@
+
+    @property
+    def preset_modes(self):
+        """Return the available presets."""
+        modes = []
+        if ThermostatEcoTrait.NAME in self._device.traits:
+            trait = self._device.traits[ThermostatEcoTrait.NAME]
+            for mode in trait.available_modes:
+                if mode in PRESET_MODE_MAP:
+                    modes.append(PRESET_MODE_MAP[mode])
+        return modes
+
+    @property
+    def fan_mode(self):
+        """Return the current fan mode."""
+        if FanTrait.NAME in self._device.traits:
+            trait = self._device.traits[FanTrait.NAME]
+            return FAN_MODE_MAP.get(trait.timer_mode, FAN_OFF)
+        return FAN_OFF
+
+    @property
+    def fan_modes(self):
+        """Return the list of available fan modes."""
+        if FanTrait.NAME in self._device.traits:
+            return list(FAN_INV_MODE_MAP.keys())
+        return []
+
+    @property
+    def supported_features(self):
+        """Bitmap of supported features."""
+        return self._supported_features
+
+    def _get_supported_features(self):
+        """Compute the bitmap of supported features from the current state."""
+        features = 0
+        if HVAC_MODE_HEAT_COOL in self.hvac_modes or HVAC_MODE_AUTO in self.hvac_modes:
+            features |= SUPPORT_TARGET_TEMPERATURE_RANGE
+        if HVAC_MODE_HEAT in self.hvac_modes or HVAC_MODE_COOL in self.hvac_modes:
+            features |= SUPPORT_TARGET_TEMPERATURE
+        if ThermostatEcoTrait.NAME in self._device.traits:
+            features |= SUPPORT_PRESET_MODE
+        if FanTrait.NAME in self._device.traits:
+            # Fan trait may be present without actually support fan mode
+            fan_trait = self._device.traits[FanTrait.NAME]
+            if fan_trait.timer_mode is not None:
+                features |= SUPPORT_FAN_MODE
+        return features
+
+    async def async_set_hvac_mode(self, hvac_mode):
+        """Set new target hvac mode."""
+        if hvac_mode not in self.hvac_modes:
+            return
+        if hvac_mode not in THERMOSTAT_INV_MODE_MAP:
+            return
+        api_mode = THERMOSTAT_INV_MODE_MAP[hvac_mode]
+        if ThermostatModeTrait.NAME not in self._device.traits:
+            return
+        trait = self._device.traits[ThermostatModeTrait.NAME]
+        await trait.set_mode(api_mode)
+
+    async def async_set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        low_temp = kwargs.get(ATTR_TARGET_TEMP_LOW)
+        high_temp = kwargs.get(ATTR_TARGET_TEMP_HIGH)
+        temp = kwargs.get(ATTR_TEMPERATURE)
+        if ThermostatTemperatureSetpointTrait.NAME not in self._device.traits:
+            return
+        trait = self._device.traits[ThermostatTemperatureSetpointTrait.NAME]
+        if self.preset_mode == PRESET_ECO or self.hvac_mode == HVAC_MODE_HEAT_COOL:
+            if low_temp and high_temp:
+                return await trait.set_range(low_temp, high_temp)
```

Comment:
Nothing is checking this return value. Please separate the await from the return and have the return statement on its own on a separate line.